src/server/pps/server/api_server.go

```diff
@@ -1181,35 +1106,45 @@ func (a *apiServer) authorizeModifyPipeline(ctx context.Context, operation pipel
 		return err
 	}
 
-	// Check that the user is authorized to write to the output repo
-	if operation == opPipelineCreate {
-		_, err = pachClient.InspectRepo(info.Pipeline.Name)
-		if err != nil && strings.HasSuffix(err.Error(), "not found") {
-			return nil // No output repo exists -- it will be created
+	// 3) Check that the user is authorized to write to the output repo.
+	// Note: authorizeModifyPipeline is called before CreateRepo creates a
+	// PipelineInfo proto in etcd, so PipelineManager won't have created an output
+	// repo yet, and it's possible to check that the output repo doesn't exist
+	// (if it did exist, we'd have to check that the user has permission to write
+	// to it, and this is simpler)
+	var required auth.Scope
+	switch operation {
+	case opPipelineCreate:
+		_, err := pachClient.WithCtx(ctx).InspectRepo(info.Pipeline.Name)
+		fmt.Printf("pachClient.InspectRepo(\"%s\") => _, %v\n", info.Pipeline.Name, err)
```

Comment:
Looks like there are a few debug statements leftover here

Reply:
Ah, whoops, thanks. Got rid of these. `git diff origin/master | grep 'fmt.Print'` didn't find any leftovers, so I think this should be good now